### PR TITLE
Add initial project structure, gradle files and workflow files

### PR DESCRIPTION
**Pull request by LakshanWeerasinghe**
## Purpose
$subject

## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: ballerina/Ballerina.toml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+[package]
+org = "ballerina"
+name = "data.yaml"
+version = "0.1.0"
+authors = ["Ballerina"]
```
Comment (SasinduDilshara) [inline ballerina/Ballerina.toml:5]: Add your name as the author

File: ballerina/Ballerina.toml [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+keywords = ["yaml"]
+repository = "https://github.com/ballerina-platform/module-ballerina-data.yaml"
+license = ["Apache-2.0"]
+distribution = "2201.8.1"
```
Comment (SasinduDilshara) [inline ballerina/Ballerina.toml:9]: Shall we update this to u9

File: ballerina/yaml_api.bal [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+        Options options = {}, typedesc<anydata> t = <>)
+        returns t|Error = @java:Method {'class: "io.ballerina.stdlib.data.yaml.Native"} external;
+        
+# Represents the YAML schema available for the parser.
+#
+# + FAILSAFE_SCHEMA - Generic schema that works for any YAML document
+# + JSON_SCHEMA - Schema supports all the basic JSON types
+# + CORE_SCHEMA - An extension of JSON schema that allows more human-readable presentation
+public enum YAMLSchema {
+    FAILSAFE_SCHEMA,
```
Comment (SasinduDilshara) [inline ballerina/yaml_api.bal:35]:
```suggestion
    FAIL_SAFE_SCHEMA,
```
wdyt?

File: gradle.properties [hunk 4]
```diff
@@ -0,0 +1,15 @@
+org.gradle.caching=true
+group=io.ballerina.stdlib
+version=0.1.0-SNAPSHOT
+ballerinaLangVersion=2201.8.1
```
Comment (SasinduDilshara) [inline gradle.properties]: We need to change this to u9

File: ballerina/build.gradle [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+            return extVersion
+        }
+    } else {
+        return extVersion.replace("${project.ext.snapshotVersion}", "")
+    }
+}
+
+apply plugin: 'io.ballerina.plugin'
+
+ballerina {
```
Comment (prakanth97) [inline ballerina/build.gradle:60]: Add 
```
testCoverageParam = "--code-coverage --coverage-format=xml --includes=io.ballerina.lib.data.*:ballerina.*"
```

File: ballerina/yaml_api.bal (hunk 3 above)
Comment (LakshanWeerasinghe) [reply ballerina/yaml_api.bal:35]: Will update in the next PR. This PR send to add the workflow and skeleton of the workflow.

File: README.md [hunk 6]
```diff
@@ -1,2 +1,81 @@
...
+
+        ./gradlew clean build -PbalJavaDebug=<port>
+
+7. Publish ZIP artifact to the local `.m2` repository:
+
+        ./gradlew clean build publishToMavenLocal
+
+8. Publish the generated artifacts to the local Ballerina central repository:
+
+        ./gradlew clean build -PpublishToLocalCentral=true
```
Comment (prakanth97) [inline README.md:61]: Add a new line here

File: README.md [hunk 7]
```diff
@@ -1,2 +1,81 @@
...
+
+## Code of conduct
+
+All contributors are encouraged to read the [Ballerina Code of Conduct](https://ballerina.io/code-of-conduct).
+
+## Useful links
+
+* Chat live with us via our [Discord server](https://discord.gg/ballerinalang).
+* Post all technical questions on Stack Overflow with the [#ballerina](https://stackoverflow.com/questions/tagged/ballerina) tag.
+* For more information go to the [`xmldata` library](https://lib.ballerina.io/ballerina/xmldata/latest).
```
Comment (prakanth97) [inline README.md]:
```suggestion
* For more information go to the [`yml` library](https://lib.ballerina.io/ballerina/yml/latest).
```

File: ballerina/build.gradle [hunk 8]
```diff
@@ -0,0 +1,119 @@
...
+    }
+}
+
+updateTomlFiles.dependsOn copyStdlibs
+
+build.dependsOn "generatePomFileForMavenPublication"
+build.dependsOn ":${packageName}-native:build"
+build.dependsOn deleteDependencyTomlFiles
+
+test.dependsOn ":${packageName}-native:build"
```
Comment (prakanth97) [inline ballerina/build.gradle:119]: Add these as well 

```

publish.dependsOn build
publishToMavenLocal.dependsOn build
```

**Review by SasinduDilshara — DISMISSED** [review]

**Comment by SasinduDilshara** [comment]
Other than existing comments, LGTM to me

**Review by prakanth97 — APPROVED** [review]

**Pull request merged**